api/src/main/java/com/velocitypowered/api/command/CommandManager.java

```diff
@@ -1,87 +1,60 @@
 package com.velocitypowered.api.command;
 
+import com.velocitypowered.api.event.command.CommandExecuteEvent;
 import java.util.concurrent.CompletableFuture;
 
 /**
- * Represents an interface to register a command executor with the proxy.
+ * Handles the registration and execution of commands.
  */
 public interface CommandManager {
 
   /**
-   * Registers the specified command with the manager with the specified aliases.
+   * Returns a builder to register a {@link LegacyCommand}.
    *
-   * @param command the command to register
-   * @param aliases the alias to use
-   *
-   * @deprecated This method requires at least one alias, but this is only enforced at runtime.
-   *             Prefer {@link #register(String, Command, String...)} instead.
+   * @return a legacy command builder
    */
-  @Deprecated
-  void register(Command command, String... aliases);
```

Comment:
Question: As this presently stands this would break the API. I assume you just want to put the API in front of users before you add any backwards compatibility?

Reply:
Yeah, this would require a major bump, so it's a long-term PR. Users are expected to not instantiate the `Command` interface/subclass it and instead use `LegacyCommand.Builder` and `BrigadierCommand.Builder` instances obtainable from the command manager.